astropy/nddata/mixins/tests/test_ndslicing.py

```diff
@@ -154,6 +154,9 @@ def test_slicing_all_npndarray_shape_diff():
     assert nd2.wcs.pixel_to_world(1) == nd.wcs.pixel_to_world(3)
 
 
+@pytest.mark.xfail(
+    sys.version_info[:2] >= (3, 12), reason="KeyError: slice(2, 5, None)"
```

Comment:
I still wish @mwcraig or @bmorris3 can look into why this test gives KeyError in Python 3.12. NDData is widely used in Jdaviz stack and wouldn't be good if we overlook a real incompatibility like this.

Reply:
Took another look at this and tbh I don't understand why it ever returned a type error....the `KeyError` is generated because the "array" in this test case is really a dictionary, and when we try to slice it that fails. In any event, I'm confident that checking for `KeyError` is the way to go.

Reply:
@mwcraig , what does that mean? The existing TypeError is a bug?